pkg/controller/plan/kubevirt.go

```diff
@@ -1279,6 +1284,67 @@ func (r *KubeVirt) vmPreference(vm *plan.VMStatus) (virtualMachine *cnv.VirtualM
 	return
 }
 
+func (r *KubeVirt) setIntanceType(vm *plan.VMStatus, object *cnv.VirtualMachine) (virtualMachine *cnv.VirtualMachine, err error) {
+	if r.Plan.Spec.SelectedInstanceType == "" {
+		return object, nil
+	}
+	intancetypeName, kind, err := r.getInstancetype(vm, r.Plan.Spec.SelectedInstanceType)
+	if err != nil {
+		return
+	}
+	object.Spec.Instancetype = &cnv.InstancetypeMatcher{Name: intancetypeName, Kind: kind}
+	virtualMachine = object
+	return
+}
+
+// Attempt to find a suitable instance type
+func (r *KubeVirt) getInstancetype(vm *plan.VMStatus, instancetypeName string) (name, kind string, err error) {
+	name, kind, err = r.getVirtualMachineInstancetype(instancetypeName)
+	if err != nil {
+		if k8serr.IsNotFound(err) {
+			r.Log.Info("could not find a local instance type for destination VM. trying cluster wide",
+				"vm",
+				vm.String())
+		} else {
+			return
+		}
+	}
+
+	name, kind, err = r.getVirtualMachineClusterInstancetype(vm, instancetypeName)
+	return
+}
+
+func (r *KubeVirt) getVirtualMachineInstancetype(instancetypeName string) (name, kind string, err error) {
+	virtualMachineClusterInstancetype := &instancetype.VirtualMachineInstancetype{}
+	err = r.Destination.Client.Get(
+		context.TODO(),
+		client.ObjectKey{Name: instancetypeName, Namespace: r.Plan.Spec.TargetNamespace},
+		virtualMachineClusterInstancetype)
+	if err != nil {
+		return
+	}
+	return instancetypeName, "VirtualMachineInstancetype", nil
```

Comment:
There could be a constant for `VirtualMachineInstancetype` in the KubeVirt API. (not 100% sure)

Reply:
yeah, I also suspect such constant would exist in the kubevirt API but I saw that such a string is used all over the place..

Reply:
https://github.com/kubevirt/kubevirt/blob/74f5cb9bcfee98661548ae5dfe900430d8cee9f9/staging/src/kubevirt.io/api/instancetype/register.go#L26